docs/sources/operators-guide/running-production-environment-with-helm/configuration-with-helm.md

```diff
@@ -0,0 +1,282 @@
+
+Decide whether you want to use a `ConfigMap` or `Secret` to store the configuration. Handling `ConfigMap` is a little bit simpler, but be ware of sensitive information.
+
+### Use external ConfigMap
+
+Prepare a `ConfigMap` object where the configuration is placed under the `mimir.yaml` data key.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-mimir-config
+data:
+  mimir.yaml: |
+    <configuration>
+```
+
+Replace `<configuration>` with the configuration as multiline text, be mindful of indentation. The name `my-mimir-config` is just an example.
+
+Set the following value for the Helm chart:
+
+```yaml
+useExternalConfig: true
+externalConfigSecretName: my-mimir-config
+externalConfigVersion: "0"
+```
+
+### Use external Secret
+
+Prepare a `Secret` object where the configuration is base64 encoded and placed under the `mimir.yaml` data key.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-mimir-config
+data:
+  mimir.yaml: <configuration-base64>
+```
+
+Replace `<configuration-base64>` with the configuration encoded as base64 format string. The name `my-mimir-config` is just an example.
+
+Set the following value for the Helm chart:
+
+```yaml
+useExternalConfig: true
+externalConfigSecretName: my-mimir-config
+configStorageType: Secret
+externalConfigVersion: "0"
+```
+
+### Update the configuration
+
+In order to make components aware of configuration changes, either:
+
+- Update the value in `externalConfigVersion` and run `helm update`
+- or restart components affected by the configuration change manually
+
+## Injecting credentials
+
+Credentials should be kept in `Secret` objects or in a credential vault. The Helm chart value `global.extraEnvFrom` can be used to inject the credentials into the runtime environment variables of the Grafana Mimir components. The data keys will become environment variables and usable in the Grafana Mimir configuration. For example `AWS_SECRET_ACCESS_KEY` can be referenced as `${AWS_SECRET_ACCESS_KEY}` in the configuration. See the [Example](#example-of-configuration-managed-with-helm) for a practical application.
+
+Grafana Mimir will not keep track of changes to the credentials. If the credentials change, Grafana Mimir services should be restarted to use the new value. An easy way to trigger such restart is to provide a `global.podAnnotation` which will be applied to all Grafana Mimir components. Changing the value of the global annotation will instruct Kubernetes to restart the components. For example changing `global.podAnnotations.bucketSecretVersion` from `'0'` to `'1'` triggers a restart - note that [pod annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) can only be strings.
+
+## Example of configuration managed with Helm
```

Comment:
Do you want to move this under `## Manage the configuration with Helm`?

Reply:
I had it there, but it was a lot, I'll talk to doc writer

Reply:
You have a reference under the Manage the configuration with Helm to the example. Because the example would be in the middle of other steps, it makes it cleaner to put the example below as one of the last section.